core/src/components/item-sliding/item-sliding.tsx

```diff
@@ -462,15 +503,62 @@ export class ItemSliding implements ComponentInterface {
 
       openSlidingItem = undefined;
       style.transform = '';
+
+      if (animationType === 'modern') {
+        this.animateSlidingItemOptions(0, rtl);
+      }
       return;
     }
     style.transform = `translate3d(${-openAmount}px,0,0)`;
+
+    if (animationType === 'modern') {
+      this.animateSlidingItemOptions(openAmount, rtl);
+    }
+
     this.ionDrag.emit({
       amount: openAmount,
       ratio: this.getSlidingRatioSync(),
     });
   }
 
+  private animateSlidingItemOptions(openAmount: number, isRTL: boolean) {
+    /**
+     * The total width of the 'processed' options.
+     * Used to calculate the offset to move each option off the
+     * screen, on top of each other.
+     */
+    let optionWidthOffset = 0;
+
+    const side = this.state === SlidingState.Start ? 'start' : 'end';
+
+    const options = Array.from(this.getOptions(side)?.querySelectorAll('ion-item-option') || []);
+    const optsWidth = this.getOptionsWidth(side);
+
+    const progress = Math.abs(openAmount / optsWidth);
+
+    options.forEach((option, index) => {
```

Comment:
The calculation doesn't seem correct when quickly dragging:


https://github.com/ionic-team/ionic-framework/assets/2721089/972bb09e-139b-4a45-96a8-e12ee5fd4d66